codegen/src/main/java/software/amazon/awssdk/codegen/docs/PaginationDocs.java

```diff
@@ -73,10 +79,52 @@ public String getDocsForSyncResponseClass(ClassName clientInterface) {
                         .toString();
     }
 
+    /**
+     * Constructs additional documentation on the async client operation that is appended to the service documentation.
+     */
+    public String getDocsForAsyncOperation() {
+        return CodeBlock.builder()
+                        .add("<p>This is a variant of {@link #$L($T)} operation. "
+                             + "The return type is a custom publisher that can be subscribed to request a stream of response "
+                             + "pages. SDK will internally handle making service calls for you.\n</p>",
+                             operationModel.getMethodName(), requestType())
+                        .add("<p>When the operation is called, an instance of this class is returned.  At this point, "
+                             + "no service calls are made yet and so there is no guarantee that the request is valid. "
+                             + "The subscribe method should be called as a request to stream data. For more info, "
+                             + "see {@link $T#$L($T)}. If there are errors in your "
+                             + "request, you will see the failures only after you start streaming the data.</p>",
+                             getPublisherType(), SUBSCRIBE_METHOD_NAME, getSubscriberType())
+                         .add(getAsyncCodeSnippets())
+                        .build()
+                        .toString();
+    }
+
+    /**
+     * Constructs javadocs for the generated response classes of a paginated operation in Async client.
+     * @param clientInterface A java poet {@link ClassName} type of the Async client interface
+     */
+    public String getDocsForAsyncResponseClass(ClassName clientInterface) {
+        return CodeBlock.builder()
+                        .add("<p>Represents the output for the {@link $T#$L($T)} operation which is a paginated operation."
+                             + " This class is a type of {@link $T} which can be used to provide a sequence of {@link $T} "
+                             + "response pages as per demand from the subscriber.</p>",
+                             clientInterface, getPaginatedMethodName(), requestType(), getPublisherType(),
+                             syncResponsePageType())
+                        .add("<p>When the operation is called, an instance of this class is returned.  At this point, "
+                             + "no service calls are made yet and so there is no guarantee that the request is valid. "
+                             + "The subscribe method should be called as a request to stream data. For more info, "
```

Comment:
Can we add some documentation on what multiple subscribes means? I.E. each one is a separate iteration from the starting request.

Reply:
Added